pkg/auth/ldaputil/client.go

```diff
@@ -78,3 +84,24 @@ func (l *LDAPClientConfig) Connect() (*ldap.Conn, error) {
 		return nil, fmt.Errorf("unsupported scheme %q", l.Scheme)
 	}
 }
+
+// Connect returns an established LDAP connection to which a bind has occurred if a bind DN and
+// password were given. If no error is returned, the caller is responsible for closing the connection
+func (l *LDAPClientConfig) Connect() (*ldap.Conn, error) {
```

Comment:
I agree, however, that `CreateConnection()` and `Connect()` are bad names.

I was thinking about `Connect()` and `ConnectAndBind()` but  I did not like `ConnectAndBind()` that would not actually do a binding if credentials were not present.